impl/src/main/java/org/ehcache/internal/sizeof/DefaultSizeOfEngineProviderFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.internal.sizeof;
+
+import org.ehcache.spi.service.ServiceCreationConfiguration;
+import org.ehcache.spi.service.ServiceFactory;
+import org.ehcache.spi.sizeof.SizeOfEngineProvider;
+
+/**
+ * @author Abhilash
+ *
+ */
+public class DefaultSizeOfEngineProviderFactory implements ServiceFactory<SizeOfEngineProvider> {
+
+  @Override
+  public SizeOfEngineProvider create(ServiceCreationConfiguration<SizeOfEngineProvider> configuration) {
+    long maxDepth = Long.MAX_VALUE;
```

Comment:
I wonder if that's the best default value. IIRC ehcache 2 used to use 100 as the default, but I'm open to any suggestion over this.